docs/guides/tfhers/shared-key.md

```diff
@@ -0,0 +1,247 @@
+# Scenario 1: Shared Secret Key
+
+The first scenario involves a single party seeking to combine both Concrete and TFHE-rs in a computation. In this case, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. It is possible to generate keys in both directions. In case 1.1, the steps are as follows:
+
+1. We do a classical keygen in Concrete, which generates a bunch of secret keys (we would be interested in the one used for TFHE-rs) and public keys
+2. We will use the secret key meant for TFHE-rs to launch a partial keygen (i.e. starting from a secret key and generating the rest of the keys) in TFHE-rs
+
+Case 1.2 is basically the opposite, the steps are as follows:
+
+1. We do a classical keygen in TFHE-rs, which generates a single secret key and different public keys
+2. We will use the secret key to launch a partial keygen in Concrete
+
+After the key generation is complete and we have both keysets, we can perform computations, encryption, and decryption on both ends.
+
+![alt text](../_static/tfhers/tfhers_uc1.excalidraw.svg)
+
+## Initial Setup and Config
+
+The first step is to define the TFHE-rs ciphertext we want to use (see [Overview](./README.md)). This includes specifying cryptographic and encoding parameters. We don't want to choose these parameters ourselves, because TFHE-rs already has a precomputed list of parameters we can use. We will be using [these ones](https://github.com/zama-ai/tfhe-rs/blob/2bd9f7aab412c92d0642a35fa87d1d0f4e2193b9/tfhe/src/shortint/parameters/classic/gaussian/p_fail_2_minus_64/ks_pbs.rs#L255) throughout this guide.
+
+In short, we will first look at what are good set of parameters we want to use in TFHE-rs, then just copy paste those in Concrete so that we have compatible ciphertexts (using the same parameters).
+
```

Comment:
The first step is to define the TFHE-rs ciphertext type that will be used in the computation. This involves specifying both cryptographic and encoding parameters. TFHE-rs provides a pre-computed list of recommended parameters, which we will use to avoid manual selection. You can find the specific parameters [here](https://github.com/zama-ai/tfhe-rs/blob/2bd9f7aab412c92d0642a35fa87d1d0f4e2193b9/tfhe/src/shortint/parameters/classic/gaussian/p_fail_2_minus_64/ks_pbs.rs#L255).

In short, we first determine a suitable set of parameters from TFHE-rs and then apply them in Concrete. This ensures that the ciphertexts generated in both systems will be compatible by using the same cryptographic parameters.